src/libraries/Common/tests/System/IO/ReadOneAtATimeStream.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.IO.Tests
+{
+    public class ReadOneAtATimeStream : Stream
+    {
+        MemoryStream _memoryStream;
+
+        public ReadOneAtATimeStream(byte[] buffer)
+        {
+            _memoryStream = new MemoryStream(buffer);
+        }
+
+        public override bool CanRead => true;
+
+        public override bool CanSeek => false;
+
+        public override bool CanWrite => false;
+
+        public override long Length => _memoryStream.Length;
+
+        public override long Position
+        {
+            get => _memoryStream.Position;
+            set => throw new NotSupportedException();
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            if (count == 0 || Position == Length)
+            {
+                return 0;
+            }
+
+            _memoryStream.ReadExactly(buffer, offset, 1);
+            return 1;
+        }
+
+        public override void Flush() { }
+        public override long Seek(long offset, SeekOrigin origin) => throw new NotSupportedException();
+        public override void SetLength(long value) => throw new NotSupportedException();
+        public override void Write(byte[] buffer, int offset, int count) => throw new NotSupportedException();
+    }
```

Comment:
Could you simplify this by just deriving from MemoryStream instead of Stream?

Reply:
We also have a DelegateStream type in tests; it lets you just construct an instance, supplying the ctor with a delegate to use for various operations, so you might not even have to declare your own type.  And if you do continue to have your own, assuming you derive from MemoryStream such that it can be really short, you could just make it a private nested class right next to the test that's using it.